tensordict/nn/probabilistic.py

```diff
@@ -56,7 +73,7 @@ class set_interaction_mode(_DecoratorContextManager):
     def __init__(self, mode: str = "mode") -> None:
         _insert_interaction_mode_deprecation_warning("set_")
         super().__init__()
-        self.mode = mode
+        self.mode = InteractionType.from_str(mode)
```

Comment:
Can I do
```
set_interaction_mode(InteractionType.MODE)
```
instead?
Shouldn't this be the "proper" way to call it?
IMO we should do
```suggestion
        if isinstance(mode, str):
            warning(...)
            self.mode = InteractionType.from_str(mode)
       else:
             self.mode = mode
```

Reply:
Yeah, I've thought about this, but since we're deprecating this fcn, any new usage should preferably be using the new alternative `set_interaction_type(InteractionType.MODE)` and all the existing usage only passes `str` arg, so didn't think changing the old interface to add this support was necessary. Unless I've missed any use cases where old fcn + new interface is somehow preferred? 🤔

Reply:
> Yeah, I've thought about this, but since we're deprecating this fcn, any new usage should preferably be using the new alternative `set_interaction_type(InteractionType.MODE)` and all the existing usage only passes `str` arg, so didn't think changing the old interface to add this support was necessary. Unless I've missed any use cases where old fcn + new interface is somehow preferred? 🤔

Oh got it, I missed that
So IIUC with `set_interaction_type` you can only pass `Enum` and with `set_interaction_mode` you can pass strings (but you'll get a warning), right?
Does the warning tells you that when changing your code to `set_interaction_mode` you should also use `Enum`?

Reply:
> So IIUC with set_interaction_type you can only pass Enum and with set_interaction_mode you can pass strings (but you'll get a warning), right?

Yes, that's right!

> Does the warning tells you that when changing your code to set_interaction_mode you should also use Enum?

The new `set_interaction_type` and `default_interaction_type` are type-hinted to `InteractionType` enum only, so linting should pick it up if one tries to feed a str to them. But I will add a note to the warning message to highlight the type change as well 👍